decryptor/postgresql/packet_handler.go

```diff
@@ -262,15 +273,60 @@ func (packet *PacketHandler) IsParse() bool {
 	return packet.messageType[0] == ParseMessageType
 }
 
-//GetParseQuery return query string from Parse packet or error
-func (packet *PacketHandler) GetParseQuery() (string, error) {
-	packet.logger.Debugln("GetParseQuery")
-	parse, err := NewParsePacket(packet.descriptionBuf.Bytes())
+// IsParseComplete return true if packet has ParseComplete type
+func (packet *PacketHandler) IsParseComplete() bool {
+	return packet.messageType[0] == ParseCompleteMessageType
+}
+
+// IsBind return true if packet has Bind type
+func (packet *PacketHandler) IsBind() bool {
+	return packet.messageType[0] == BindMessageType
+}
+
+// IsBindComplete return true if packet has BindComplete type
+func (packet *PacketHandler) IsBindComplete() bool {
+	return packet.messageType[0] == BindCompleteMessageType
+}
+
+// IsExecute return true if packet has Execute type
+func (packet *PacketHandler) IsExecute() bool {
+	return packet.messageType[0] == ExecuteMessageType
+}
+
+// GetParseData returns parsed Parse packet data.
+// Use this only if IsParse() is true.
```

Comment:
imho better to not make it as contract for callee and check one more time here with `IsParse`

Reply:
> imho better to not make it as contract for callee and check one more time here with IsParse

Well, it's not like a strict contract... If this is no Parse packet, `GetParseData` will return `nil` and an error. It can be used to check for packet type but I feel that having separate `IsParse` is cleaner, and it's a cheap check. That way you can tell a difference between a malformed Parse packet and not a Parse packet at all.